File: 03-poo/10-miembros-de-clase-y-de-instancia/java/soluciones/RetoContadorUsuarios.java
public class RetoContadorUsuarios {
    static class Usuario {
        private static int total;
        Usuario(){ total++; }
        static int totalCreados(){ return total; }
    }
    public static void main(String[] args){ new Usuario(); new Usuario(); new Usuario(); System.out.println(Usuario.totalCreados()); }
}

File: 03-poo/10-miembros-de-clase-y-de-instancia/scala/soluciones/RetoContadorUsuarios.scala
class UsuarioReto:
  UsuarioReto.incrementar()
object UsuarioReto:
  private var total = 0
  private def incrementar(): Unit = total += 1
  def totalCreados: Int = total

@main def retoContadorUsuarios(): Unit =
  new UsuarioReto; new UsuarioReto; new UsuarioReto
  println(UsuarioReto.totalCreados)
